### PR TITLE
Adding MacOS-12 to CI

### PR DESCRIPTION
**Pull request by aprsa**
This PR adds macos-12 to the CI matrix.

### CODE REVIEW

File: .github/workflows/on_pr_test_python_31x.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
   pull_request:
   push:
     branches:
+      - 'macos12_ci'
```
Comment (kecnry) [inline .github/workflows/on_pr_test_python_31x.yml:9]: This PR itself should still run CI without needing to list it in the list of push branches, right?

```suggestion
```

File: .github/workflows/on_pr_test_python_38.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ on:
   pull_request:
   push:
     branches:
+      - 'macos12_ci'
```
Comment (kecnry) [inline .github/workflows/on_pr_test_python_38.yml:9]:
```suggestion
```

File: .github/workflows/on_pr_test_python_39.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ on:
   pull_request:
   push:
     branches:
+      - 'macos12_ci'
```
Comment (kecnry) [inline .github/workflows/on_pr_test_python_39.yml:9]:
```suggestion
```

**Review by kecnry — COMMENTED** [review]
This shouldn't currently cause conflicts with the bugfix release, so is probably fine to merge directly into master.  We'll only need to rebase the bugfix branch if we are trying to address the failure there.

**Comment by aprsa** [comment]
Just for the record, I canceled all tests for the last 3 commits because the changes were inconsequential.

**Pull request merged**